test/assert_is_void_test.cpp

```diff
@@ -0,0 +1,108 @@
+//
+//  assert_is_void_test.cpp - tests BOOST_ASSERT_IS_VOID
+//
+//  Copyright (c) 2015 Ion Gaztanaga
+//
+// Distributed under the Boost Software License, Version 1.0.
+// See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt
+//
+
+#include <boost/config.hpp>
+
+// default case, !NDEBUG
+// BOOST_ASSERT(x) -> assert(x)
+
+#undef NDEBUG
+#include <boost/assert.hpp>
+
+#ifdef BOOST_ASSERT_IS_VOID
+#error "Error: BOOST_ASSERT should be void in NDEBUG"
```

Comment:
Right. It should read:

"BOOST_ASSERT should NOT be void if NDEBUG is not defined"